map.go

```diff
@@ -717,13 +754,34 @@ func (mi *MapIterator) Next(keyOut, valueOut interface{}) bool {
 		return false
 	}
 
+	// gessing a random start key
+	if !supportNullKey.value && mi.randomStartGuess {
```

Comment:
I *think* this will cause the first entry to be missed by iteration, because `mi.prevKey` is not reset after these checks, and it has already been iterated on once.

Reply:
I'm not sure to understand (the reset after check) ... but `mi.prevKey` is used between map.Next() call as context, `mi.prevKey` is updated line 796 in normal behaviour ; and `mi.prevKey` is used at the initial key (nil in recent kernel)